Makefile

```diff
@@ -1,18 +1,52 @@
-BINS = archive2disk cexec grub2disk kexec oci2disk qemuimg2disk rootio slurp syslinux writefile
+# Use bash
+SHELL       := bash
+.SHELLFLAGS := -o pipefail -euc
+.ONESHELL:
 
-include rules.mk
+# Second expansion is used by the image targets to depend on their respective binaries. It is
+# necessary because automatic variables are not set on first expansion.
+# See https://www.gnu.org/software/make/manual/html_node/Secondary-Expansion.html.
+.SECONDEXPANSION:
 
-all: $(BINS)
+# Define the list of actions that can be built.
+ACTIONS := archive2disk cexec grub2disk kexec oci2disk qemuimg2disk rootio slurp syslinux writefile
 
-.PHONY: $(BINS)
-$(BINS):
+# Define the commit for tagging images.
+GIT_COMMIT := $(shell git rev-parse HEAD)
+
+# Define Docker registry details.
```

Comment:
nit: `docker` -> `container`

Reply:
I'm in the habbit of naming it after the protocol, which has changed to the 'distribution' specification.